test/e2e/proxy_test.go

```diff
@@ -231,14 +233,114 @@ func TestProxyFlow(t *testing.T) {
 				require.NoError(t, err)
 				require.NotEmpty(t, createdApp)
 				assert.Equal(t, expectedApp.Spec.DisplayName, createdApp.Spec.DisplayName)
-			})
+			}) // end of successful workspace context
 
 			t.Run("invalid workspace context", func(t *testing.T) {
 				proxyWorkspaceURL := hostAwait.ProxyURLWithWorkspaceContext("notexist")
 				// Start a new websocket watcher which watches for Application CRs in the user's namespace
 				_, err := hostAwait.CreateAPIProxyClient(t, user.token, proxyWorkspaceURL)
-				require.EqualError(t, err, `an error on the server ("unable to get target cluster: the requested space in not available") has prevented the request from succeeding`)
+				require.EqualError(t, err, `an error on the server ("unable to get target cluster: the requested space is not available") has prevented the request from succeeding`)
 			})
+
+			t.Run("successful space lister request", func(t *testing.T) {
+				// given
+				proxyCl, err := hostAwait.CreateAPIProxyClient(t, user.token, hostAwait.APIProxyURL)
+				require.NoError(t, err)
+				expectedWorkspace := commonproxy.NewWorkspace(user.compliantUsername,
+					commonproxy.WithNamespaces([]toolchainv1alpha1.SpaceNamespace{
+						{
+							Name: user.compliantUsername + "-tenant",
+							Type: "default",
+						},
+					}),
+					commonproxy.WithOwner(user.signup.Name),
+					commonproxy.WithRole("admin"),
+				)
+
+				t.Run("successful list", func(t *testing.T) {
+					// when
+					workspaces := &toolchainv1alpha1.WorkspaceList{}
+					err = proxyCl.List(context.TODO(), workspaces)
+
+					// then
+					require.NoError(t, err)
+					require.Len(t, workspaces.Items, 1)
```

Comment:
sWe need to cover the case when there is actually some workspace sharing and the list of available workspaces contains more than one workspace.

Let's make sure we test a real life scenarios.
Create multiple spaces & spacebindings. One user has access to a single workspace only. And at least two more users which share their spaces with each other. Then test that:
- each user gets expected list of spaces
- doesn't get the workspaces which the user doesn't have access to included into the result (so no noise workspaces included)
- same for GET specific workspace request.

Reply:
@alexeykazakov e2e tests have been updated